components/server/ee/src/prebuilds/prebuilt-status-maintainer.ts

```diff
@@ -243,14 +229,19 @@ export class PrebuildStatusMaintainer implements Disposable {
         const ctx = TraceContext.childContext("periodicUpdatableCheck", {});
 
         try {
-            const unresolvedUpdatables = await this.workspaceDB.trace(ctx).getUnresolvedUpdatables();
+            const unresolvedUpdatables = await this.workspaceDB.trace(ctx).getUnresolvedUpdatables(MAX_UPDATABLES);
             for (const updatable of unresolvedUpdatables) {
-                if (Date.now() - Date.parse(updatable.workspace.creationTime) > MAX_UPDATABLE_AGE) {
+                const age = Date.now() - Date.parse(updatable.workspace.creationTime);
+                if (age > MAX_UPDATABLE_AGE && age < IGNORED_UPDATEABLE_AGE) {
```

Comment:
:pray: 

Did you think about going without `IGNORED_UPDATEABLE_AGE` because we now should be marking all entries as "resolved"? I feel that would be a bit "cleaner".
Both would work, obviously.

Reply:
That's a second barrier, to ensure we're not overloading. This is also ensures an immediate fix of the issue, no db migration required.

Reply:
> This is also ensures an immediate fix of the issue, no db migration required.

Wouldn't that be the case as well if we didn't have it? It would doUpdate all entries, and mark them as resolved?

Reply:
There is difference though. 

Without this kind of hard limit here, `getUnresolvedUpdatables` will continue to return VERY OLD stuff for which this updates will be tried to get applied, i.e. tons of unnecessary/failing commit status updates via GH App. THIS isn't what we want, do we?

Reply:
I feel having the DB in a good shape is beneficial. If you think it takes too long, and we should include a migration, I'd rather do that then skipping old entries.
Having systems that automatically converge is always easier to reason about then time-window stuff.